### DISCUSSION TITLE
Sending custom events to ElasticSearch

### DISCUSSION THREAD

**Discussion by WojSad** [magma/magma#11805] [Q&A]
Hi all. Please kindly advise on the following:

Me and my team need to integrate Domain Proxy with Elasticsearch.
We want to be able to **send custom messages** there and then display them on a newly developed UI page. We also want to be able to **filter search results on fields that these custom messages will have**. 

For example: 

Let's say our message looks like this: `{"some_key1": "foo", "some_key2": "bar"}`. We want to be able to search and filter on some_key1 and some_key2. 

As far as I understand this is not possible with current eventd-driven pipeline, as Events described here https://docs.magmacore.org/docs/lte/eventd#how-to-create-an-event have a specific structure where the actual payload of the message goes to a text field named value. 
If we put our payload `{"some_key1": "foo", "some_key2": "bar"}` in the value field, it will get flattened to a string and searching by its keys will become quite complex. 

Here are a few Ideas on how to push these new events to ES and make them filterable and searchable: 

- Create our own module that will push custom structured messages to fluentd to a different index (say domain_proxy) and then simply query Elasticsearch using whatever filters we want. However in this scenario we wouldn't be using fluent-bit as Domain Proxy is not a part of AGW and thus we would need to send them directly on Fluentd's http endpoint. Fluentd in turn would have to act as parser for these events in order to properly insert them to ES.

- Use the current pipeline and construct a query that will parse the `value` text field on the fly during querying and search by its keys. Unfortunately we weren't able to find a way to do it without using full text search on the string.

- Use the current pipeline and push Events to fluentd, but change fluentd's config so that it would redirect these events to a different index and create new fields on messages using json processor. 

Please let me know if any of these approaches are ok or perhaps if there is another way to do it.
Thanks

**Comment by Neudrino** [comment]
Maybe someone from @magma/approvers-domain-proxy can help out here.
